README
NAME
    Dist::Zilla::Plugin::Test::UnusedVars - Release tests for unused
    variables

VERSION
    version 2.000006

SYNOPSIS
    In "dist.ini":

        [Test::UnusedVars]

    Or, give a list of files to test:

        [Test::UnusedVars]
        file = lib/My/Module.pm
        file = bin/verify-this

DESCRIPTION
    This is an extension of Dist::Zilla::Plugin::InlineFiles, providing the
    following file:

        xt/release/unused-vars.t - a standard Test::Vars test

AVAILABILITY
    The project homepage is
    <http://metacpan.org/release/Dist-Zilla-Plugin-Test-UnusedVars/>.

    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit <http://www.perl.com/CPAN/> to find a
    CPAN site near you, or see
    <https://metacpan.org/module/Dist::Zilla::Plugin::Test::UnusedVars/>.

SOURCE
    The development version is on github at
    <http://github.com/doherty/Dist-Zilla-Plugin-Test-UnusedVars> and may be
    cloned from
    <git://github.com/doherty/Dist-Zilla-Plugin-Test-UnusedVars.git>

BUGS AND LIMITATIONS
    You can make new bug reports, and view existing ones, through the web
    interface at
    <https://github.com/doherty/Dist-Zilla-Plugin-Test-UnusedVars/issues>.

AUTHORS
    *   Marcel Grünauer <marcel@cpan.org>

    *   Mike Doherty <doherty@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2010 by Mike Doherty.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Perl extension {{$dist->name}}

{{$NEXT}}
  - Remove requirement for Moose::Autobox in tests [kentnl, GH #6]

2.000006  2014-04-29
  - Allow specifying a list of files to test manually

2.000005  2013-08-03
  - Don't shoot yourself in the foot when testing [kentnl, GH #4]

2.000004  2013-04-09
  - Test only for our own deprecation warnings

2.000003  2012-06-08
  - Use a traditional decimal version [dolmen, GH #2]

v2.0.2    2012-05-21
  - Declare Dist::Zilla dependency
  - Use a strict version number [dolmen, GH #2]

2.0.1     2011-09-07
  - Packaging updates

v2.0.0    2011-09-07
  - Rename to Dist::Zilla::Plugin::Test::UnusedVars

1.100860  2010-03-27 23:36:51 Europe/Vienna
  - original version


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.015.
use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "Release tests for unused variables",
  "dist_author" => [
    "Marcel Gr\x{fc}nauer <marcel\@cpan.org>",
    "Mike Doherty <doherty\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-Test-UnusedVars",
  "dist_version" => "2.000006",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Plugin::Test::UnusedVars",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Data::Section" => 0,
    "Dist::Zilla" => 4,
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Moose" => 0,
    "Path::Tiny" => 0,
    "Test::More" => "0.96",
    "Test::Vars" => "0.005",
    "namespace::autoclean" => 0,
    "perl" => "5.008",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Moose::Autobox" => 0,
    "Test::DZil" => 0,
    "Test::More" => "0.96",
    "Test::Output" => 0,
    "autodie" => 0
  }
);


my %fallback_build_requires = (
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Module::Build" => "0.3601",
  "Moose::Autobox" => 0,
  "Test::DZil" => 0,
  "Test::More" => "0.96",
  "Test::Output" => 0,
  "autodie" => 0
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.015.
use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Release tests for unused variables",
  "AUTHOR" => "Marcel Gr\x{fc}nauer <marcel\@cpan.org>, Mike Doherty <doherty\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Module::Build" => "0.3601"
  },
  "CONFIGURE_REQUIRES" => {
    "Module::Build" => "0.3601"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Test-UnusedVars",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::Plugin::Test::UnusedVars",
  "PREREQ_PM" => {
    "Data::Section" => 0,
    "Dist::Zilla" => 4,
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Moose" => 0,
    "Path::Tiny" => 0,
    "Test::More" => "0.96",
    "Test::Vars" => "0.005",
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Moose::Autobox" => 0,
    "Test::DZil" => 0,
    "Test::More" => "0.96",
    "Test::Output" => 0,
    "autodie" => 0
  },
  "VERSION" => "2.000006",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Data::Section" => 0,
  "Dist::Zilla" => 4,
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Role::TextTemplate" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Module::Build" => "0.3601",
  "Moose" => 0,
  "Moose::Autobox" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::More" => "0.96",
  "Test::Output" => 0,
  "Test::Vars" => "0.005",
  "autodie" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



